src/qudi/hardware/switches/power_switch_keysight.py

```diff
@@ -0,0 +1,181 @@
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with qudi.
+If not, see <https://www.gnu.org/licenses/>.
+"""
+
+try:
+    import visa
+except ImportError:
+    import pyvisa as visa
+from qudi.core.configoption import ConfigOption
+from qudi.core.statusvariable import StatusVar
+from qudi.util.mutex import RecursiveMutex
+from qudi.interface.switch_interface import SwitchInterface
+
+
+class PowerSwitch(SwitchInterface):
+    """ This class is implements communication with the Keysight E36234A power supply using pyVISA
+
+    Example config for copy-paste:
+
+    power_switch3:
+        module.Class: 'switches.power_switch_keysight.PowerSwitch'
+        options:
+            interface: 'USB0::0x2A8D::0x3402::MY59001224::INSTR'
+            hardware_name: 'Keysight_Power2'  # optional
+            switch_time: 1  # optional
+            remember_states: False  # optional
+            switches:
+                APD: ['OFF', 'ON']  # optional
+                Amplifier: ['OFF', 'ON']  # optional
+    """
+
+    # customize available switches in config. Each switch needs a tuple of at least 2 state names.
+    _switches = ConfigOption(name='switches', missing='error')
+    # optional name of the hardware
+    _hardware_name = ConfigOption(name='hardware_name', default='power_supply', missing='nothing')
+    # if remember_states is True the last state will be restored at reloading of the module
+    _remember_states = ConfigOption(name='remember_states', default=False, missing='nothing')
+    # switch_time to wait after setting the states for the solenoids to react
+    _switch_time = ConfigOption(name='switch_time', default=0.5, missing='nothing')
+    # name of the serial interface where the hardware is connected.
+    # Use e.g. the Keysight IO connections expert to find the device.
+    serial_interface = ConfigOption('interface', 'USB0::0x2A8D::0x3402::MY59001224::INSTR')
+
+    # StatusVariable for remembering the last state of the hardware
+    _states = StatusVar(name='states', default=None)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.lock = RecursiveMutex()
+        self._resource_manager = None
+        self._instrument = None
+        self._switches = self._chk_refine_available_switches(self._switches)
+
+    def on_activate(self):
+        """ Prepare module, connect to hardware.
+        """
+
+        self._resource_manager = visa.ResourceManager()
+        self._instrument = self._resource_manager.open_resource(self.serial_interface)
+        # set the maximum/protection output Voltage/current of the power supply
+        # Excelitas APD: 5V,1.3A
+        self._instrument.write('INST CH1')
+        self._instrument.write('OUTP 0')
+        self._instrument.write('VOLT 5')

```

Comment:
Maximum voltages should be ConfigOptions, not hardcoded.

Reply:
Now the voltages and currents are taken from `ConfigOption`